fetch.bs

```diff
@@ -1561,19 +1567,15 @@ whose <a for=request>destination</a> is "<code>audio</code>", "<code>audioworkle
 "<code>script</code>", "<code>style</code>", "<code>track</code>", "<code>video</code>",
 "<code>xslt</code>", or the empty string.
 
-<p>A <dfn export>potential-navigation-or-subresource request</dfn> is a
-<a for=/>request</a> whose
-<a for=request>destination</a> is
-"<code>object</code>" or "<code>embed</code>".
-
 <p>A <dfn export>non-subresource request</dfn> is a <a for=/>request</a>
-whose <a for=request>destination</a> is "<code>document</code>",
-"<code>report</code>", "<code>serviceworker</code>", "<code>sharedworker</code>",
-or "<code>worker</code>".
+whose <a for=request>destination</a> is "<code>document</code>", "<code>embed</code>",
+"<code>frame</code>", "<code>iframe</code>", "<code>object</code>", "<code>report</code>",
+"<code>serviceworker</code>", "<code>sharedworker</code>", or "<code>worker</code>".
 
 <p>A <dfn export>navigation request</dfn> is a <a for=/>request</a> whose
 <a for=request>destination</a> is
-"<code>document</code>".
+"<code>document</code>", "<code>embed</code>", "<code>frame</code>", "<code>iframe</code>",
+or "<code>object</code>".
```

Comment:
Does this work for the uses in https://w3c.github.io/ServiceWorker/? Also, do these always have a reserved client and replaces client id? I don't think so right? All of that is rather messy at the moment I suppose and embed/object are somewhat poorly understood, so maybe waggling in the right direction is good enough.

cc @jakearchibald @jungkees

Reply:
1. Didn't we decide that `object` and `embed` requests bypassed Service Workers?

2. Assuming it's not a no-op from that perspective, it looks like "navigation request" is used three times in `ServiceWorker`. Perhaps @jakearchibald and @jungkees could weigh in on the latter two, as I'm not sure I understand what they're trying to do:

    * 12.3 of https://w3c.github.io/ServiceWorker/#on-fetch-request-algorithm (which is a no-op, as it will never match "the navigation triggering it was initiated with a shift+reload or equivalent")
    * 12.7 of https://w3c.github.io/ServiceWorker/#on-fetch-request-algorithm 
    * 19.9 of https://w3c.github.io/ServiceWorker/#terminate-service-worker-algorithm

Reply:
1. Sounds agreeable to me, but would be good to have tests for that, especially where embed/object is being navigated.

Reply:
1. Yes `object` and `embed` return early in [Handle Fetch](https://w3c.github.io/ServiceWorker/#handle-fetch) step 11.
1. I think all the three cases are OK: 1) is force-reload as @mikewest said 2) is checking if the request is for main resource for navigation preload 3) is to fill up fetchEvent.replacesClientId for navigation requests. Fetch requests went through [process a navigate fetch](https://html.spec.whatwg.org/#process-a-navigate-fetch) should have a replaces client id at this point.